openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx

```diff
@@ -74,13 +80,22 @@ const PermissionProvider: FC<PermissionProviderProps> = ({ children }) => {
     return AppState.getCurrentUserDetails();
   }, [AppState.userDetails, AppState.nonSecureUserDetails]);
 
+  const handledVerifiedUser = () => {
+    const urlPathname = cookieStorage.getItem(REDIRECT_PATHNAME);
+    if (urlPathname) {
+      cookieStorage.removeItem(REDIRECT_PATHNAME);
+      history.push(urlPathname);
+    }
+  };
+
   /**
    * Fetch permission for logged in user
    */
   const fetchLoggedInUserPermissions = async () => {
     try {
       const response = await getLoggedInUserPermissions();
       setPermissions(getUIPermission(response.data || []));
+      handledVerifiedUser();
```

Comment:
Do we need this in the permission provider?

Reply:
@Sachin-chaurasiya Yes, as for redirecting user in a Protected route `LoggedInUserPermissions` should be present. If not, user will redirect user to 404 after login which is not expected.